h/indexer.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+
+from h.celery import celery
+
+from h.api import storage
+from h.api.search.index import index
+from h.api.search.index import delete
+
+__all__ = (
+    'add_annotation',
+    'delete_annotation',
+)
+
+
+@celery.task
+def add_annotation(id_):
+    annotation = storage.fetch_annotation(celery.request, id_)
```

Comment:
Ah, ok. This is a good idea, but you're going to need to be a bit careful if you do this. Because the view function is called inside a transaction, that means that you're going to be calling `add_annotation.delay(id)` while the transaction is still open.

While not likely in practice, that means that in theory that message could race to rabbit, get dispatched to a worker, and then do a `SELECT` against the database before the transaction is committed. Which would be Bad (:tm:).

So, while this is absolutely fine, you'll need to change either the subscriber to trigger the job in a [response callback](http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/hooks.html#using-response-callbacks). 

In fact, we should probably trigger all `AnnotationEvents` inside response callbacks (being sure not to do so if the response is an error), so perhaps it would be better to do it once in `h.api.views`. This would ensure that:

- `AnnotationEvents` can't modify annotations, as they don't have access to the session
- they only get fired once the data is successfully written to the database
- errors in `AnnotationEvent` subscribers don't cause the transaction to abort (which, given [we've already got code to ensure this in one case](https://github.com/hypothesis/h/blob/1b184e98549010a86646a2dea9ba66b8a50ae969/h/subscribers.py#L53-L63), seems like the right thing to do)